twill/utils.py

```diff
@@ -128,6 +128,17 @@ def find_link(self, pattern: str) -> Optional[Link]:
             if regex.search(link.text) or regex.search(link.url):
                 return link
         return None
+    
+    def find_links(self, pattern: str) -> Optional[List[Link]]:
+        """Find all links with a given pattern on the result page."""
+        regex = re.compile(pattern)
+        links = []
+        for link in self.links:
```

Comment:
A list comprehension would be easier here.

Reply:
Fixed